### PR TITLE
feat: support setting time range in Copy From statement

### LINKED ISSUE #3511: Support setting time range in `Copy From` statement
### What type of enhancement is this?

User experience

### What does the enhancement do?

As title. 

Currently there's `timestamp_range: Option<TimestampRange>` field in the `struct CopyTableRequest`, but it's not set. We can set it from the SQL parser first, then check if it works.

The `copy from` in postgresql supports "where condition" (https://www.postgresql.org/docs/current/sql-copy.html), maybe we can do that, too. For this issue, we can extract the time range from the "where condition", and leave more general filtering in the future.

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by realtaobo**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
Resolve #3511 
## What's changed and what's your intention?

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
Reuse the time range specified in the WITH clause and ensure the data can be filtered based on the 'ts' column.  Hence, `Copy table x from datasource with (start_time='2022-06-15 07:02:37', end_time='2022-06-15 07:02:38');` statement should work properly.
- How does this PR work? Need a brief introduction for the changed logic (optional)
All work based [filter_record_batch](https://docs.rs/datafusion/latest/datafusion/common/arrow/compute/fn.filter_record_batch.html) in file:https://github.com/GreptimeTeam/greptimedb/blob/2ae2a6674e8a780cb93723de9b749b65ea28562e/src/common/recordbatch/src/adapter.rs#L94-L95

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/common/query/src/logical_plan/expr.rs [hunk 1]
```diff
@@ -18,6 +18,33 @@ use common_time::Timestamp;
 use datafusion_common::{Column, ScalarValue};
 use datafusion_expr::expr::Expr;
 use datafusion_expr::{and, binary_expr, Operator};
+use datatypes::data_type::DataType;
+use datatypes::schema::ColumnSchema;
+use datatypes::value::Value;
+
+pub fn build_same_type_ts_filter(
```
Comment (killme2008) [inline src/common/query/src/logical_plan/expr.rs:25]: Could you add a comment about this function? Thank you. It's better to have a comment for every public function.

File: src/common/recordbatch/src/filter.rs [hunk 2]
```diff
@@ -144,13 +151,42 @@ impl SimpleFilterEvaluator {
...
+pub fn batch_filter(
+    batch: &RecordBatch,
+    predicate: &Arc<dyn PhysicalExpr>,
+) -> DfResult<RecordBatch> {
+    predicate
+        .evaluate(batch)
+        .and_then(|v| v.into_array(batch.num_rows()))
+        .and_then(|array| {
+            let filter_array = match as_boolean_array(&array) {
+                Ok(boolean_array) => Ok(boolean_array.to_owned()),
```
Comment (killme2008) [inline src/common/recordbatch/src/filter.rs:164]: Prefer to use `clone` to indicate the operation explicitly.
```suggestion
                Ok(boolean_array) => Ok(boolean_array.clone()),
```

**Review by killme2008 — APPROVED** [review]
Almost LGTM. Please take a look @fengjiachun @WenyXu

**Review by WenyXu — APPROVED** [review]
Nice done🚀

**Pull request merged**